setup.py

```diff
@@ -162,7 +163,7 @@ def run(self):
 with open('README.rst', 'r') as readme_file:
     longdesc = readme_file.read()
 
-setup(
+__name__ == '__main__' and setup(
```

Comment:
* I don't think we'll ever want to import setup.py as a script but if we do, we shouldn't execute any code so the guard should include more of setup.py
* Explicit is better than implicit so if we do this we should use ```if``` rather than ```and```.

With those two thoughts in mind, if we do put a guard in, we probably want to put the code that "does things immediately" (and is not involved in the import process) into a function and use the traditional ```if __name__ == '__main__': main_function()``` idiom to call it.  (So from line 133)

Reply:
Ageed. I know that we don't do this. But I've seen this happening when pytest looks up for tests (when it's not configure to target/ignore things), so I might imagine that someone might run into it.